lucene/core/src/java/org/apache/lucene/search/TopScoreDocCollector.java

```diff
@@ -276,7 +283,20 @@ protected void updateMinCompetitiveScore(Scorable scorer) throws IOException {
           && pqTop.score != Float.NEGATIVE_INFINITY) { // -Infinity is the score of sentinels
       // since we tie-break on doc id and collect in doc id order, we can require
       // the next float
-      scorer.setMinCompetitiveScore(Math.nextUp(pqTop.score));
```

Comment:
Good idea -- I was toying around with the same (in fact, the first iteration of the original PR had that, but then I removed it since I was not sure). Updated now, thanks.